### ISSUE TITLE
Feature request: Python: kill job step

### ISSUE THREAD

**Issue by jameshcorbett** [flux-framework/flux-core#2893]
I've been updating my python bindings implementation, and one feature that would be nice to have in the bindings is the ability to kill job steps manually. I know my users are interested in that feature, and I can currently implement it in the Flux CLI by just killing the ``mini run`` process.

I think it could be useful for killing specific applications if they hang or their performance degrades. In ML-driven workflows I have even heard some (very tentative) talk of killing applications based on their intermediate results.

If you like the idea, it might be nice if the ``flux.job.wait`` would return killed job steps. I would like to have one thread sleeping on ``flux.job.wait`` and a second thread looking around for steps to kill, and it would be nice if the first thread could respond to the actions of the second.

**Comment by grondo** [comment]
There is already a `flux_job_kill(3)` C API function we could trivially wrap in the Python job class. (I gather you could use it currently via cffi, but I'll have to experiment a bit to remember how to do that)

 `flux.job.wait` should return for waitable jobs, no matter how they terminate, so I think it will work as you desire.

**Comment by garlick** [comment]
Bindings for job `kill`, `cancel`, and `raise` RPCs are indeed missing and should be added.  Thanks!

> If you like the idea, it might be nice if the flux.job.wait would return killed job steps.

It does - or I'm misunderstanding your request.  We don't really have a "job step" concept in flux.  There are flux instances and jobs running within them.  `wait` works on jobs.  If you want to wait on a job in a sub-instance, then the python program would have to connect to that instance to wait on its jobs.  Does that make sense?

**Comment by garlick** [comment]
Oops sorry for the duplicate response!

**Comment by jameshcorbett** [comment]
Thanks @garlick  and @grondo, sounds great. And yeah, @garlick, by "job step" I just meant "job," (hard to keep the terminology straight sometimes) I just wanted to check that the interaction between `wait` and `kill` would work as I hoped.
